stubs/docker/docker/types/services.pyi

```diff
@@ -1,170 +1,185 @@
 from _typeshed import Incomplete
+from typing import Literal, TypeVar, overload
+
+from .healthcheck import Healthcheck
+
+_T = TypeVar("_T")
 
 class TaskTemplate(dict[str, Incomplete]):
     def __init__(
         self,
-        container_spec,
-        resources: Incomplete | None = None,
-        restart_policy: Incomplete | None = None,
-        placement: Incomplete | None = None,
-        log_driver: Incomplete | None = None,
-        networks: Incomplete | None = None,
-        force_update: Incomplete | None = None,
+        container_spec: ContainerSpec,
+        resources: Resources | None = None,
+        restart_policy: RestartPolicy | None = None,
+        placement: Placement | list[str] | None = None,
+        log_driver: DriverConfig | None = None,
+        networks: list[str | NetworkAttachmentConfig] | None = None,
+        force_update: int | None = None,
     ) -> None: ...
     @property
-    def container_spec(self): ...
+    def container_spec(self) -> ContainerSpec: ...
     @property
-    def resources(self): ...
+    def resources(self) -> Resources: ...
     @property
-    def restart_policy(self): ...
+    def restart_policy(self) -> RestartPolicy: ...
     @property
-    def placement(self): ...
+    def placement(self) -> Placement: ...
 
 class ContainerSpec(dict[str, Incomplete]):
     def __init__(
         self,
-        image,
-        command: Incomplete | None = None,
-        args: Incomplete | None = None,
-        hostname: Incomplete | None = None,
-        env: Incomplete | None = None,
-        workdir: Incomplete | None = None,
-        user: Incomplete | None = None,
-        labels: Incomplete | None = None,
-        mounts: Incomplete | None = None,
-        stop_grace_period: Incomplete | None = None,
-        secrets: Incomplete | None = None,
-        tty: Incomplete | None = None,
-        groups: Incomplete | None = None,
-        open_stdin: Incomplete | None = None,
-        read_only: Incomplete | None = None,
-        stop_signal: Incomplete | None = None,
-        healthcheck: Incomplete | None = None,
-        hosts: Incomplete | None = None,
-        dns_config: Incomplete | None = None,
-        configs: Incomplete | None = None,
-        privileges: Incomplete | None = None,
-        isolation: Incomplete | None = None,
-        init: Incomplete | None = None,
-        cap_add: Incomplete | None = None,
-        cap_drop: Incomplete | None = None,
-        sysctls: Incomplete | None = None,
+        image: str,
+        command: str | list[str] | None = None,
+        args: list[str] | None = None,
+        hostname: str | None = None,
+        env: dict[str, Incomplete] | list[str] | None = None,
```

Comment:
```suggestion
	# Any: str | bytes | None, but dict is invariant
        env: dict[str, Any] | list[str] | None = None,
```

Alternatively you could make a union of all the possible dict value type union combination.

Or leave it as `Incomplete` and a problem for someone else to solve :P